flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/calcite/FlinkTypeFactory.scala

```diff
@@ -190,11 +190,11 @@ class FlinkTypeFactory(typeSystem: RelDataTypeSystem) extends JavaTypeFactoryImp
     buildRelDataType(
       tableSchema.getFieldNames.toSeq,
       tableSchema.getFieldTypes map {
-        case TimeIndicatorTypeInfo.PROCTIME_INDICATOR
-          if isStreaming.isDefined && !isStreaming.get =>
+        case tp: TimeIndicatorTypeInfo
+          if !tp.isEventTime && isStreaming.isDefined && !isStreaming.get =>
           InternalTypes.TIMESTAMP
-        case TimeIndicatorTypeInfo.ROWTIME_INDICATOR
-          if isStreaming.isDefined && !isStreaming.get =>
+        case tp: TimeIndicatorTypeInfo
+          if tp.isEventTime && isStreaming.isDefined && !isStreaming.get =>
```

Comment:
Combine the above two case in to `case tp: TimeIndicatorTypeInfo if isStreaming.isDefined && !isStreaming.get` ?